29983 - Add New Registrar Info and Signature

meawong opened the pull request.

severinbeauvais reviewed (approved):
LGTM.

Did you find the API version file? Re: https://business-ar-api-dev-dy4loprnwa-nn.a.run.app/v1/meta/info

Were you able to test this?

severinbeauvais commented:
> Did you find the API version file? Re: https://business-ar-api-dev-dy4loprnwa-nn.a.run.app/v1/meta/info

See https://github.com/bcgov/business-ar/blob/f6be594d5a6efff1328fe6e04e75a77f7113f384/api/pyproject.toml#L3

severinbeauvais commented on api/src/business_ar_api/utils/registrar_metadata.py:
```diff
@@ -57,6 +57,14 @@ class RegistrarInfo:  # pylint: disable=too-few-public-methods
             "signatureImage": "registrar_signature_5.png",
             "signatureImageAndText": "registrar_signature_and_text_5.png",
             "startDate": "2025-04-18T00:00:00",
+            "endDate": "2025-08-11T23:59:59"
```
Not an issue.

severinbeauvais commented on api/src/business_ar_api/utils/registrar_metadata.py:
```diff
@@ -57,6 +57,14 @@ class RegistrarInfo:  # pylint: disable=too-few-public-methods
+            "title": "Registrar of Companies",
+            "signatureImage": "registrar_signature_6.png",
+            "signatureImageAndText": "registrar_signature_and_text_6.png",
+            "startDate": "2025-08-12T00:00:00",
```
Wait! There's a request from the business to defer this. Olga is negotiating the exact requirement. Hence, do not merge this PR yet.

meawong commented:
Tested locally:

Updated Annual Report `filing_date `to Aug 11 and see Sinead's signature:

<img width="809" height="692" alt="image" src="https://github.com/user-attachments/assets/02db40b2-2982-4e98-974e-beff3195a937" />

Updated Annual Report `filing_date `to Aug 12 and see Kerry's Signature:
<img width="782" height="696" alt="image" src="https://github.com/user-attachments/assets/b610acf0-38f0-44a1-a28b-bb18ec8b26a4" />
